src/projections.F90

```diff
@@ -500,9 +520,17 @@ subroutine point_volume(x0, tu, tv, tw, ku, kv, kw, coef, nctlu, nctlv, nctlw, n
 
         !Check that this is a descent direction -
         !otherwise use the negative gradient
-        if (pgrad >= 0.0) then
-            update = -grad / NORM2(grad)
+        if (pgrad .gt. 0.0) then
+            update = -grad / (NORM2(grad) + eps)
```

Comment:
Not just stability, w/o that eps check, the step can be zero and that division results in a NaN. I am thinking the exit criteria based on the conditions we discussed today will render that unnecessary, but I would still like to have it since this is one location a NaN might show up in the computation. If the step gets screwed up large due to division by zero, the linesearch just backtracks until its tiny anyways.